KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs

```diff
@@ -173,7 +183,91 @@ internal object GetContentItemModel(Type t, JToken item, JToken modularContent,
                 }
             }
 
+            foreach (var property in richTextPropertiesToBeProcessed)
+            {
+                var value = property.GetValue(instance);
+                var propValue = ((JObject)item["elements"]).Properties()
+                    ?.FirstOrDefault(p => PropertyMapper.IsMatch(property, p.Name, system?.Type))
+                    ?.FirstOrDefault()["value"];
+
+                var modularContentInRichText =
+                                ((JObject)propValue?.Parent?.Parent)?.Property("modular_content")?.Value;
+
+                var currentlyProcessedString = new RichTextContentItem()
+                {
+                    ContentItemCodeName = system.Codename,
+                    RichTextElementCodeName = property.Name
+                };
+                if (currentlyResolvedRichStrings.Contains(currentlyProcessedString))
+                {
+                    value = RemoveContentItemsFromRichText((string) value);     //In case we've stumbled upon an element which is already being processed, we need to use it as 
+                                                                                //is (therefore removing content items) to prevent circular dependency
+                }
+                else
+                {
+                    currentlyResolvedRichStrings.Add(currentlyProcessedString);
+                    value = ProcessContentItemsInRichText(modularContent, processedItems, (string) value,
+                        modularContentInRichText, currentlyResolvedRichStrings);
+                    currentlyResolvedRichStrings.Remove(currentlyProcessedString);
+                }
+                if (value != null)
+                {
+                    property.SetValue(instance, value);
+                }
+
+            }
+
             return instance;
         }
+
+        private object ProcessContentItemsInRichText(JToken modularContent, Dictionary<string, object> processedItems, string value, JToken modularContentInRichText, HashSet<RichTextContentItem> currentlyResolvedRichStrings)
+        {
+            var usedCodenames = Newtonsoft.Json.JsonConvert.DeserializeObject<IEnumerable<string>>(modularContentInRichText.ToString());
+            var contentItemsInRichText = new Dictionary<string, object>();
+                
+            foreach (var codenameUsed in usedCodenames)
+            {
+                object contentItem;
+                if (processedItems.ContainsKey(codenameUsed) && currentlyResolvedRichStrings.All(x => x.ContentItemCodeName != codenameUsed))   // This is to reuse content items which were processed already, but not those 
```

Comment:
1) "they  are may contain"
2) please put the comments on a new line before the `if`...(+adjust other comments in this way as well)